### PR TITLE
angular-karma

### PR DESCRIPTION
**Pull request by ankit-dq**

### CODE REVIEW

File: Node/angular-karma/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+module.exports = function (config) {
+  config.set({
+    basePath: '',
+    frameworks: ['jasmine', '@angular-devkit/build-angular'],
+    plugins: [
```
Comment (michael-siek) [inline Node/angular-karma/karma.conf.js:8]: I don't think we need to specify these in here. @straker Thoughts?

File: Node/angular-karma/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      require('karma-jasmine'),
+      require('karma-chrome-launcher'),
+      require('karma-jasmine-html-reporter'),
+      require('karma-coverage'),
+      require('@angular-devkit/build-angular/plugins/karma'),
+      require('@axe-devtools/karma-json-reporter'),
+      // require('@axe-devtools/reporter')
```
Comment (michael-siek) [inline Node/angular-karma/karma.conf.js:15]:
```suggestion
```

File: Node/angular-karma/package.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "angular-proj1",
```
Comment (michael-siek) [inline Node/angular-karma/package.json:2]:
```suggestion
  "name": "angular-karma",
```
Or something a little bit more descriptive

File: Node/angular-karma/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  "version": "0.0.0",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "rimraf rimraf ./a11y-results && ng test"
```
Comment (michael-siek) [inline Node/angular-karma/package.json:9]:
```suggestion
    "pretest": "rimraf ./a11y-results"
    "test": "ng test"
```

File: Node/angular-karma/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "install": "^0.13.0",
+    "jasmine-core": "~4.3.0",
+    "jquery": "^3.6.1",
+    "karma": "~6.4.0",
+    "karma-chrome-launcher": "~3.1.0",
+    "karma-coverage": "~2.2.0",
+    "karma-jasmine": "~5.1.0",
+    "karma-jasmine-html-reporter": "~2.0.0",
+    "node-polyfill-webpack-plugin": "^2.0.1",
+    "npm": "^9.6.5",
```
Comment (michael-siek) [inline Node/angular-karma/package.json:49]: Is this required?

File: Node/angular-karma/src/app/alert-details/alert-details.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+// import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (michael-siek) [inline Node/angular-karma/src/app/alert-details/alert-details.component.spec.ts:1]: Is this file needed if its commented out?

File: Node/angular-karma/src/app/app.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+});
+
+
+  it(`should have as title 'angular_proj1'`, () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect(app.title).toEqual('angular_proj1');
+  });
+
+  // it('should render title', () => {
```
Comment (michael-siek) [inline Node/angular-karma/src/app/app.component.spec.ts:44]: Same here
> Are these LOC required? If not lets remove it

File: Node/angular-karma/src/app/product-shop/product-shop.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+describe('ProductShopComponent', () => {
+  let component: ProductShopComponent;
+  let fixture: ComponentFixture<ProductShopComponent>;
+  let reporter:any;
+  
+  
+
+  beforeEach(async () => {
+    // axeDevTools.init();
+    // reporter = new Reporter('A11yResults', './a11y-results');
```
Comment (michael-siek) [inline Node/angular-karma/src/app/product-shop/product-shop.component.spec.ts:13]:
```suggestion
```

File: Node/angular-karma/src/app/product-shop/product-shop.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  //   axeDevTools.run(component).then((results:any) => {
+  //     console.log(results);
+    
+  //  });
```
Comment (michael-siek) [inline Node/angular-karma/src/app/product-shop/product-shop.component.spec.ts:29]:
```suggestion
```

**Review by michael-siek — CHANGES_REQUESTED** [review]
inline comments. Also if its not on here we should run [prettier](https://prettier.io/) on these example projects

File: Node/angular-karma/package.json (hunk 5 above)
Comment (ankit-dq) [reply Node/angular-karma/package.json:50]: Karma-jasmin and karma-jasmine are required.

Do not need the other 2 i.e.
    "node-polyfill-webpack-plugin": "^2.0.1",
    "npm": "^9.6.5",
Hence removing them.

File: Node/angular-karma/karma.conf.js (hunk 1 above)
Comment (ankit-dq) [reply Node/angular-karma/karma.conf.js:8]: The tests are not running correctly and are running into errors, If I remove this from the karma.conf.js file.

Comment (straker) [reply Node/angular-karma/karma.conf.js:8]: My guess is that since the `@angular-devkit/build/angular/plugins/karma` doesn't [start with `karma-*`](https://karma-runner.github.io/2.0/config/plugins.html) then it isn't loaded automatically. If you even have to specify a plugin to load, then you have to specify all of them.

Comment (ankit-dq) [reply Node/angular-karma/karma.conf.js:8]: I did. This is how the whole piece looks like.

`  config.set({
    basePath: '',
    frameworks: ['jasmine', '@angular-devkit/build-angular'],
    plugins: [
      require('karma-jasmine'),
      require('karma-chrome-launcher'),
      require('karma-jasmine-html-reporter'),
      require('karma-coverage'),
      require('@angular-devkit/build-angular/plugins/karma'),
      require('@axe-devtools/karma-json-reporter'),
    ]`

**Review by michael-siek — APPROVED** [review]
